lib/appscale_tools.py

```diff
@@ -653,3 +672,48 @@ def upload_app(cls, options):
       shutil.rmtree(file_location)
 
     return (login_host, http_port)
+
+  @classmethod
+  def upgrade(cls, options):
+    """ Upgrades the deployment to the latest AppScale version as well as
+    updates the data.
+
+    Args:
+      options: A Namespace that has fields for each parameter that can be
+        passed in via the command-line interface.
+    Returns:
+      A tuple containing the host and port where the application is serving
+        traffic from.
+    """
+    if os.path.exists(LocalState.get_secret_key_location(options.keyname)):
+      response = raw_input("AppScale needs to be down for this upgrade."
+        " Are you sure you want to proceed? (Y/N) ")
+      if response not in ['y', 'yes', 'Y', 'YES']:
+        raise AppScaleException("Cancelled AppScale upgrade.")
+      else:
+        AppScaleLogger.log("Shutting down AppScale...")
+        cls.terminate_instances(options)
+
+    AppScaleLogger.log("Upgrading AppScale to the latest version on "
+      "these machines: {}".format(options.ips))
+    AppScaleLogger.warn(("Running bootstrap on the machines to fetch latest " + \
+      "code and build AppScale. This will atleast take a few minutes."))
+    for ip in options.ips:
+      try:
+        RemoteHelper.ssh(ip, options.keyname, cls.BOOTSTRAP_COMMAND, options.verbose)
+        AppScaleLogger.success("Successfully upgraded AppScale to its latest version.")
+      except ShellException:
+        AppScaleLogger.warn("Error executing bootstrap command to upgrade AppScale.")
+
+    zookeeper_ips = ""
+    for zk_ip in options.zk_ips:
+      zookeeper_ips += zk_ip + " "
+
+    upgrade_script_zk_loc = cls.UPGRADE_SCRIPT + " " + zookeeper_ips
+    AppScaleLogger.log("Upgrade script {}".format(upgrade_script_zk_loc))
```

Comment:
Since this prints out the command itself, should it be a debug message?
If you want you can leave the log and only print the zookeeper_ips that are being upgraded.